wcfsetup/install/files/lib/system/bbcode/AttachmentBBCode.class.php

```diff
@@ -55,162 +62,156 @@ public function getParsedTag(array $openingTag, $content, array $closingTag, BBC
             }
         }
 
-        /** @var Attachment $attachment */
-        $attachment = MessageEmbeddedObjectManager::getInstance()->getObject(
-            'com.woltlab.wcf.attachment',
-            $attachmentID
-        );
-        if ($attachment === null) {
-            if (self::$attachmentList !== null) {
-                $attachments = self::$attachmentList->getGroupedObjects(self::$objectID);
-                if (isset($attachments[$attachmentID])) {
-                    $attachment = $attachments[$attachmentID];
+        if ($thumbnail === false) {
```

Comment:
I find this confusing, because `$thumbnail` is guaranteed to be a bool. 
```suggestion
        if (!$thumbnail) {
```

Reply:
This was intentional, because I misread that condition as `if ($thumbnail)` more than once in the original code.

However, it might be better to simply swap places of both code paths so that the first path will output the thumbnail.